File: python/authdog/exceptions.py
"""Custom exceptions for Authdog SDK."""


class AuthdogError(Exception):
    """Base exception for all Authdog SDK errors."""
    pass


class AuthenticationError(AuthdogError):
    """Raised when authentication fails."""
    pass


class APIError(AuthdogError):
    """Raised when API requests fail."""

    def __init__(self, message: str, status_code=None):
        super().__init__(message)
        self.status_code = status_code

File: planned/dart/lib/exceptions.dart
class AuthdogException implements Exception {
  final String message;
  const AuthdogException(this.message);

  @override
  String toString() => message;
}

class AuthenticationException extends AuthdogException {
  const AuthenticationException(String message) : super(message);
}

class ApiException extends AuthdogException {
  const ApiException(String message) : super(message);
}
